contracts/transfer/circuits/src/gadgets/preimage.rs

```diff
@@ -7,26 +7,39 @@
 use dusk_plonk::prelude::*;
 use plonk_gadgets::AllocatedScalar;
 use poseidon252::sponge::sponge::sponge_hash_gadget;
+use std::convert::TryInto;
 
 /// Prove knowledge of the preimage of a note,
 /// used as input for a transaction.
 #[allow(non_snake_case)]
 pub fn input_preimage(
     composer: &mut StandardComposer,
+    note_type: AllocatedScalar,
     value_commitment_x: AllocatedScalar,
     value_commitment_y: AllocatedScalar,
-    pos: AllocatedScalar,
     pk_r_x: AllocatedScalar,
     pk_r_y: AllocatedScalar,
+    randomness_x: AllocatedScalar,
+    randomness_y: AllocatedScalar,
```

Comment:
Why don't you expect a `plonk::ecc::Point` instead of this. Will make your code much easier and you have access to it's coordinates so that you can re-use them without breaking any linking.

It would be something like:
```rust
use dusk_plonk::constraint_system::ecc::Point as PlonkPoint;
pub fn input_preimage(
     composer: &mut StandardComposer;
     note_type: AllocatedScalar,
     value_commitment: PlonkPoint,
     pkr: PlonkPoint,
     randomness: PlonkPoint,
     // Whatever else....
) { 
     // Use the coordinates of the `PlonkPoint` which is a `JubJubAffine` 
     // kinda off "allocated" and so you don't break any linking.
}